contrib/gcs_importer.py

```diff
@@ -0,0 +1,203 @@
+formatter = logging.Formatter("%(asctime)s [%(levelname)s] %(message)s")
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
+
+def download_from_gcs(gcs_base_path, filename):
+    """Download file from Google Cloud Storage (GCS).
+
+    Args:
+        gcs_base_path: (str) GCS bucket path
+        filename: (str) Filename of the file to download
+
+    Returns:
+        (str) Path to downloaded file
+    """
+    storage_client = storage.Client(args.project)
+    bucket = storage_client.get_bucket(args.bucket)
+    gcs_full_path = os.path.join(gcs_base_path, filename)
+    local_path = os.path.join(args.output, filename)
+    blob = bucket.blob(gcs_full_path)
+    blob.download_to_filename(local_path)
+    logger.info('Downloaded file from GCS: {}'.format(local_path))
+    return local_path
+
+
+def setup_sketch(timeline_name, index_name, username, sketch_id=None):
+    """Use existing sketch or create a new sketch.
+
+    Args:
+        timeline_name: (str) Name of the Timeline
+        index_name: (str) Name of the index
+        username: (str) Who should own the timeline
+        sketch_id: (str) Optional sketch_id to add timeline to
+
+    Returns:
+        (str) Sketch ID
+    """
+    with app.app_context():
+        user = User.get_or_create(username=username)
+        sketch = None
+
+        if sketch_id:
+            try:
+                sketch = Sketch.query.get_with_acl(sketch_id, user=user)
+                logger.info('Using existing sketch: {} ({})'.format(
+                    sketch.name, sketch.id))
+            except Forbidden:
+                pass
+
+        if not sketch:
+            # Create a new sketch.
+            sketch_name = 'Turbinia: {}'.format(timeline_name)
+            sketch = Sketch(
+                name=sketch_name, description=sketch_name, user=user)
+            # Need to commit here to be able to set permissions later.
+            db_session.add(sketch)
+            db_session.commit()
+            sketch.grant_permission(permission='read', user=user)
+            sketch.grant_permission(permission='write', user=user)
+            sketch.grant_permission(permission='delete', user=user)
+            sketch.status.append(sketch.Status(user=None, status='new'))
+            db_session.add(sketch)
+            db_session.commit()
+            logger.info('Created new sketch: {} ({})'.format(
+                sketch.name, sketch.id))
+
+        searchindex = SearchIndex.get_or_create(
+            name=timeline_name, description='Created by Turbinia.', user=user,
+            index_name=index_name)
+        searchindex.grant_permission(permission='read', user=user)
+        searchindex.grant_permission(permission='write', user=user)
+        searchindex.grant_permission(permission='delete', user=user)
+        searchindex.set_status('processing')
+        db_session.add(searchindex)
+        db_session.commit()
+
+        if sketch.has_permission(user, 'write'):
```

Comment:
Maybe instead of not creating a timeline here when we don't have access we could create it and not associate it with the sketch?